lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -714,6 +719,8 @@ data TxSkeleton = TxSkeleton
     , txInputCount :: !Int
     , txOutputs :: ![TxOut]
     , txChange :: ![Set AssetId]
+    , txScripts :: [Script KeyHash]
+    , txMintAssets :: [AssetId]
```

Comment:
There is some somewhat confusing terminology here. `cardano-node` just refers to the minting and burning field as "mint". So this refers to both minted and burned assets.

When constructing a transaction manually, mints look something like `+3 deadbeef.aaaaaa`, whereas burns look something like `-3 deadbeef.aaaaaa`. If an asset is both minted and burned, it will just be added twice to the `txMintAssets` list. This is what we want because the Tx construction will do that same, add some data for the mint, add some data for the burn. From the perspective of tx size estimation, negative (burned) and positive (minted) quantities are no different, they are both represented with an `int64`. We don't have to do any fancy logic here, mints and burns are both treated as (qty, policyId, assetName), as they will be in the final Tx.

Reply:
I've improved the terminology and added a comment to explain the usage of the field: c0de116.

Reply:
https://github.com/input-output-hk/cardano-wallet/pull/2741/files#diff-6218e625cd8c0f10e842ef1c10eb04bebf307ccecad42038e3b12b287a763690R721